src/pages/Wallet/SendPage.tsx

```diff
@@ -17,22 +17,63 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
 import { Send } from 'lucide-react'
-import { useContext, useState } from 'react'
+import { ChangeEvent, useContext, useState } from 'react'
 import { useHistory } from 'react-router'
 import styled, { useTheme } from 'styled-components'
 
 import { GlobalContext } from '../../App'
 import { Button } from '../../components/Buttons'
+import ExpandableSection from '../../components/ExpandableSection'
 import InfoBox from '../../components/InfoBox'
 import Input from '../../components/Inputs/Input'
 import { ModalContext } from '../../components/Modal'
 import { Section } from '../../components/PageComponents/PageContainers'
 import Spinner from '../../components/Spinner'
 import { checkAddressValidity } from '../../utils/addresses'
 import { getHumanReadableError } from '../../utils/api'
-import { convertToQALPH } from '../../utils/numbers'
+import { MINIMAL_GAS_AMOUNT, MINIMAL_GAS_PRICE } from '../../utils/constants'
+import { abbreviateAmount, convertToQALPH } from '../../utils/numbers'
 import { WalletContext } from './WalletRootPage'
 
+const onAmountInputValueChange = ({
+  amount,
+  minAmount,
+  stateSetter,
+  errorMessage,
+  currentErrorState,
+  errorStateSetter,
+  shouldConvertToQALPH
+}: {
+  amount: string
+  minAmount: bigint
+  errorMessage: string
+  stateSetter: (v: string) => void
+  currentErrorState: string
+  errorStateSetter: (v: string) => void
+  shouldConvertToQALPH?: boolean
+}) => {
+  let amountNumber
+
+  try {
+    amountNumber = BigInt(shouldConvertToQALPH ? convertToQALPH(amount) : amount)
+  } catch (e) {
+    console.log(e)
+    return
+  }
+
+  if (amountNumber < minAmount) {
+    errorStateSetter(errorMessage)
+  } else {
+    if (currentErrorState) errorStateSetter('')
+  }
+
+  stateSetter(amount)
+}
+
+const getExpectedFee = (gasAmount: string, gasPriceInALPH: string) => {
```

Comment:
I would move this utility function at the end of the file (before the styling part) so that the first JS code the reader reads after the imports is the actual component, and not utility functions.

Reply:
Agreed. I'll move the `onAmountInputValueChange` as well, as explained in the other comment.

Reply:
Oops, spoke before reading this XD